### PR TITLE
Add helper for generating addict routes

### PR DESCRIPTION
**Pull request by trestrantham**
Adds `addict :routes` helper to the parent applications router.

By default, `addict :routes` will generate the following routes:

```
        register_path  POST  /register          Addict.Controller.register/2
           login_path  POST  /login             Addict.Controller.login/2
          logout_path  POST  /logout            Addict.Controller.logout/2
recover_password_path  POST  /recover_password  Addict.Controller.recover_password/2
  reset_password_path  POST  /reset_password    Addict.Controller.reset_password/2
```

But the helper also accepts overrides for `path` and/or `controller`/`action` so that:

``` elixir
addict :routes,
  logout: [path: "/sign-out", controller: ExampleApp.UserController, action: :sign_out],
  recover_password: "/password/recover",
  reset_password: "/password/reset"
```

generates:

```
        register_path  POST  /register          Addict.Controller.register/2
           login_path  POST  /login             Addict.Controller.login/2
          logout_path  POST  /sign-out          ExampleApp.UserController.sign_out/2
recover_password_path  POST  /password/recover  Addict.Controller.recover_password/2
  reset_password_path  POST  /password/reset    Addict.Controller.reset_password/2
```

### CODE REVIEW

File: example_app/priv/repo/migrations/20150221030445_add_users_table.exs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    create table(:users) do
+      add :email,         :string, size: 200
+      add :username,      :string, size: 200
+      add :hash,          :string, size: 130
+      add :recovery_hash, :string, size: 130
+
+      timestamps
+    end
+
+    create index(:users, [:email], unique: true)
```
Comment (trenpixster) [inline example_app/priv/repo/migrations/20150221030445_add_users_table.exs:14]: Nice :+1:

File: example_app/web/router.ex [hunk 2]
```diff
@@ -18,10 +19,9 @@ defmodule ExampleApp.Router do
...
 
-    post "/register", Addict.Controller, :register
-    post "/logout", ExampleApp.UserController, :logout
-    post "/login", Addict.Controller, :login
-    post "/password_recover", Addict.Controller, :recover_password
-    post "/password_reset", Addict.Controller, :reset_password
+    addict :routes,
+      logout: [path: "/logout", controller: ExampleApp.UserController, action: :signout],
+      recover_password: "/password/recover",
+      reset_password: "/password/reset"
```
Comment (trenpixster) [inline example_app/web/router.ex:25]: Can you also update [`index.html.eex`](https://github.com/trenpixster/addict/blob/master/example_app/web/templates/page/index.html.eex) with the new routes? :smile:

**Comment by trenpixster** [comment]
It'd be great to also expose this on the `README.md` on the routes configuration section.

**Comment by trestrantham** [comment]
I've already started another branch to create some view scaffolding for the 5 actions that will generate default views tied to the routes configured via `addict :routes` (a la https://github.com/etufe/phoenix_generator).

The plan is to create an `addict` folder in `web/templates` to house these 5 templates with a shared view at `web/views/addict.ex`. The templates would be normal `form`s rather than submitting via javascript.

Updates to the view/templates were planned for that branch but I can definitely update `index.html.eex` here in the meantime if you'd like. I will go ahead and update the README for the route-specific changes here.

I would love your thoughts on this general direction. Specifically in case you were thinking of going in a different direction with the views.

Cheers!

**Comment by trenpixster** [comment]
@trestrantham I was thinking in adding something like that, the objective would be to have Addict devs up and running in the shortest number of steps possible. Generating the views is an awesome improvement, it fits my objective on #9 too :+1: Care to open a new issue addressing the views generation so we can discuss there?

In the mean time I think this PR, as it is, is already a good improvement, let's merge it. If you have a chance to update `index.html.eex` it'd be great :smile:

**Comment by trestrantham** [comment]
Updated `index.html.eex` and `UserController` to reflect the updated example routes and created #24 to track template/view generation. :+1:

**Comment by trenpixster** [comment]
Thanks!

**Pull request merged**